### PR TITLE
TASK_1

### PR DESCRIPTION
**Pull request by mrSemkin**
Add task 1

### CODE REVIEW

File: task_1/task_1.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+            user_input = input('How many page download(max 500): ')
+            try:
+                n = int(user_input)
+            except ValueError:
+                print("Bad input.\nPlease enter a valid integer.")
+                i += 1
+            else:
+                return n
+        print('Too many input errors\n------ Good by. ------')
+        exit()
```
Comment (vloooo) [inline task_1/task_1.py]: you too severe to your user.
could be rewritten.
let's consider together.

File: task_1/task_1.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+    @staticmethod
+    def __input_numder_page():
+        i = 0
+        while i < 10:
+            user_input = input('How many page download(max 500): ')
+            try:
+                n = int(user_input)
+            except ValueError:
+                print("Bad input.\nPlease enter a valid integer.")
```
Comment (vloooo) [inline task_1/task_1.py]: double-quotes

File: task_1/task_1.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+                    time.sleep(10)
+                    continue
+                elif response.status_code == 404:
+                    print('Error HTTP: Page not found.')
+                break
+            else:
+                print(i)
+                i += 1
+                for film in response.json()["results"]:
+                    self.full_data.append(film)
```
Comment (vloooo) [inline task_1/task_1.py]: absolutely unreadable.
you need to add some blank lines to separate logical blocks of code

File: task_1/task_1.py [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+                        'is over the allowed limit of (40) per second.\nPlease wait 10s.')
+                    time.sleep(10)
+                    continue
+                elif response.status_code == 404:
+                    print('Error HTTP: Page not found.')
+                break
+            else:
+                print(i)
+                i += 1
+                for film in response.json()["results"]:
```
Comment (vloooo) [inline task_1/task_1.py]: double-quotes

File: task_1/task_1.py [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+                    print(
+                        f'response=429\nYour request count ({i}) '
+                        'is over the allowed limit of (40) per second.\nPlease wait 10s.')
+                    time.sleep(10)
+                    continue
+                elif response.status_code == 404:
+                    print('Error HTTP: Page not found.')
+                break
+            else:
+                print(i)
```
Comment (vloooo) [inline task_1/task_1.py]: forget about printing results in your projects.
write as mature programmer.
to see the results during development use a debugger

File: task_1/task_1.py [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+
+        if file_path:
+            if not os.path.exists(file_path):
+                os.makedirs(file_path)
+            file_path += '\\'
+
+        return file_path, file_name
+
+    def __add_file_extention(self, name):
+        f_name = name if (name.find('.csv', -4)) != -1 else name + '.csv'
```
Comment (vloooo) [inline task_1/task_1.py]: controversial 
you could use .endwith() method
and what will you do if someone decides to pass you file with extension .txt?

File: task_1/task_1.py [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+        return f_name
+
+
+    # 2.	Give a user all data
+    def get_full_data(self, f_name=None):  # Give a user all data
+
```
Comment (vloooo) [inline task_1/task_1.py]: unnecessary blank line

**Review by vloooo — COMMENTED** [review]
So, I would advise you to make a working session together to fully discuss your code. 
please contact me and will arrange time

**Comment by mrSemkin** [comment]
I'm always ready)
Set anytime.

**Pull request merged**